pkg/plugins/driverutil/reflect.go

```diff
@@ -0,0 +1,83 @@
+package driverutil
+
+import (
+	reflect "reflect"
+	"sync"
+	"time"
+
+	corev1 "github.com/rancher/opni/pkg/apis/core/v1"
+	protoreflect "google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func GetRevisionFieldIndex[T Revisioner]() int {
+	var revision corev1.Revision
+	revisionFqn := revision.ProtoReflect().Descriptor().FullName()
+	var t T
+	fields := t.ProtoReflect().Descriptor().Fields()
+	for i := 0; i < fields.Len(); i++ {
+		field := fields.Get(i)
+		if field.Kind() == protoreflect.MessageKind {
+			if field.Message().FullName() == revisionFqn {
+				return i
+			}
+		}
+	}
+	panic("revision field not found")
```

Comment:
This would always be a bug, but I can add the "bug:" prefix that I usually include for these panics

Reply:
Actually this might not be a "bug" in the same sense. If this line ever gets hit, you have some serious problems. In order to satisfy the interface 
```go
type Revisioner interface {
    proto.Message
    GetRevision() *corev1.Revision
}
```
The message would _have_ to have a revision field. For all intents and purposes this is unreachable.